homeassistant/components/apple_tv/__init__.py

```diff
@@ -255,7 +254,7 @@ async def _connect_loop(self):
         while self.is_on and self.atv is None:
             await self.connect_once(raise_missing_credentials=False)
             if self.atv is not None:
-                break
+                break  # type: ignore[unreachable]
```

Comment:
```suggestion
                # Calling self.connect_once may set self.atv, mypy does not know this
                break  # type: ignore[unreachable]
```

Reply:
Wouldn't a
```py
if TYPE_CHECKING:
    assert self.atv
```
be nicer?

Reply:
I changed the code like this, but it makes no sense; we can't assert `self.atv` because the entire point of the check is that `self.atv` may or may not be `None`